pytket/extensions/cutensornet/mps/mps_gate.py

```diff
@@ -167,14 +167,14 @@ def _apply_2q_gate(self, positions: tuple[int, int], gate: Op) -> MPSxGate:
         S_d = cp.empty(new_dim, dtype=self._real_t)
 
         # Create tensor descriptors
-        T_desc = T.get_tensor_descriptor(self._libhandle)
-        L_desc = L.get_tensor_descriptor(self._libhandle)
-        R_desc = R.get_tensor_descriptor(self._libhandle)
+        T_desc = T.get_tensor_descriptor(self._lib)

```

Comment:
not `self._lib.handle`?

Reply:
The code is correct, but I agree it might be misleading. Here, `self._lib` is the `CuTensorNetHandle` object, which wraps the actual handle on a context manager. In reality, all that this `get_tensor_descriptor` method needs is the `handle` field of this `self._lib`, but passing the `CuTensorNetHandle` achieves two things:

- I can test for `is_destroyed` inside the `get_tensor_descriptor` method.
- I can give an explicit type to the function; otherwise I'd need to use `Any` or do some ugly (and perhaps unsafe) casting to `int`.

As a rule of thumb, all of the API functions I wrote should receive a `CuTensorNetHandle`, so they are passed `self._lib`. But cuQuantum API needs the handle itself, so I pass `self._lib.handle` to them.

Reply:
Ah, I didn't realise you also tweaked `get_tensor_descriptor` - tbh I think github is being dodgy indeed...